src/srs-filter.c

```diff
@@ -317,14 +367,28 @@ xxfi_srs_milter_envrcpt(SMFICTX* ctx, char** argv) {
            cd->num, cd->state, argv[0]);
 
   // get recipient address
-  char *recip = (char *) malloc(strlen(argv[0])-1);
+  char *recip = (char *) malloc(strlen(argv[0])+1);
```

Comment:
is using strnlen here safer? (I'm not sure if useful for argv). And, can you add a comment describing why you need the extra 2 chars?